requirements-base.txt

```diff
@@ -71,6 +70,7 @@ sentry-sdk>=0.13.5
 setproctitle>=1.1.7,<1.2.0
 simplejson>=3.2.0,<3.9.0
 six>=1.10.0,<1.11.0
+social-auth-app-django==3.1.0
```

Comment:
@joshuarli How did you decide this is the library we need to use?

Reply:
Because it is.

```
social-auth-app-django==3.1.0
  - six [required: Any, installed: 1.10.0]
  - social-auth-core [required: >=1.2.0, installed: 3.2.0]
    - oauthlib [required: >=1.0.3, installed: 3.1.0]
    - PyJWT [required: >=1.4.0, installed: 1.5.3]
    - python-openid [required: >=2.2.5, installed: 2.2.5]
    - requests [required: >=2.9.1, installed: 2.20.1]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.8, installed: 2.7]
      - urllib3 [required: >=1.21.1,<1.25, installed: 1.24.2]
    - requests-oauthlib [required: >=0.6.1, installed: 1.2.0]
      - oauthlib [required: >=3.0.0, installed: 3.1.0]
      - requests [required: >=2.0.0, installed: 2.20.1]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.8, installed: 2.7]
        - urllib3 [required: >=1.21.1,<1.25, installed: 1.24.2]
    - six [required: >=1.10.0, installed: 1.10.0]
```

Reply:
lol....it would be good to provide at least _some_ explanation of how we got here. I trust you it's the right library but we should at least document `why` we picked it.

Reply:
On the upstream side of things, django-social-auth's since been replaced with [https://python-social-auth.readthedocs.io/en/latest/](https://python-social-auth.readthedocs.io/en/latest/): 

> The initial codebase is derived from django-social-auth with the idea of generalizing the process to suit the different frameworks around, providing the needed tools to bring support to new frameworks.

python-social-auth for our use case involves two components: [https://github.com/python-social-auth/social-core](https://github.com/python-social-auth/social-core) which contains the authentication backends we use, and the Django framework compatibility layer: [https://github.com/python-social-auth/social-app-django](https://github.com/python-social-auth/social-app-django).